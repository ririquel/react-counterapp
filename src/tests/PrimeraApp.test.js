import "@testing-library/jest-dom";
import React from "react";
import { shallow } from "enzyme";

const { render } = require("@testing-library/react");
const { default: PrimeraApp } = require("../PrimeraApp");

describe("Pruebas de <PrimeraApp />", () => {
  //   test('Debe de mostrar el mensaje "Hola soy Richi"', () => {
  //     const saludo = "Hola soy Richi";
  //     const { getByText } = render(<PrimeraApp saludo={saludo} />);
  //     // console.log(saludo);
  //     expect(getByText(saludo)).toBeInTheDocument();
  //   });

  test("Debe de mostrar <PrimeraApp /> correctamente", () => {
    const saludo = "Hola soy Richi";
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);
    console.log(saludo);
    expect(wrapper).toMatchSnapshot();
  });
  test("Debe de mostrar el subtitulo enviado por props", () => {
    const saludo = "Hola soy Richi";
    const subTitulo = "Soy un subtitulo";
    const wrapper = shallow(
      <PrimeraApp saludo={saludo} subtitulo={subTitulo} />
    );
    console.log(subTitulo);

    const textoParrafo = wrapper.find("p").text();

    expect(textoParrafo).toBe(subTitulo);
  });
});
