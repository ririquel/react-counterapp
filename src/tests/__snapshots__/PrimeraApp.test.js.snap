// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <PrimeraApp /> Debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola soy Richi
    !!!
  </h1>
  <p>
    Soy un subtitulo.
  </p>
</Fragment>
`;
